proof/proof_test.go

```diff
@@ -246,3 +248,64 @@ func Example_or2() {
 	// 000000b0  4d 97 a9 bf 1a 28 27 6d  3b 71 04 e1 c0 86 96 08  |M....('m;q......|
 	// Proof verified.
 }
+
+func BenchmarkProof(b *testing.B) {
+	rand := blake2xb.New([]byte("random"))
+	suites := []struct {
+		Suite
+	}{
+		{edwards25519.NewBlakeSHA256Ed25519()},
+		{curve25519.NewBlakeSHA256Curve25519(false)},
+		{curve25519.NewBlakeSHA256Curve25519(true)},
+		{nist.NewBlakeSHA256P256()},
+		{nist.NewBlakeSHA256QR512()},
+	}
+
+	for _, suite := range suites {
+		P := suite.Point().Null()
+
+		sval := map[string]kyber.Scalar{}
+		pval := map[string]kyber.Point{}
+		predicateBuilder := make([]string, 0)
+
+		for i := 0; i < b.N; i++ {
+			s := suite.Scalar().Pick(rand)
+			index := strconv.Itoa(i)
+
+			publicPoint := suite.Point().Mul(s, nil)
+
+			sval["x"+index] = s
+			predicateBuilder = append(predicateBuilder, "x"+index)
+			predicateBuilder = append(predicateBuilder, "B"+index)
+			pval["B"+index] = suite.Point().Base()
+
+			P = suite.Point().Add(P, publicPoint)
+		}
+
+		pval["P"] = P
+
+		var proof []byte
+		var err error
+		var pred Predicate
+
+		b.Run(suite.String()+"/ProofBuild", func(b *testing.B) {
+			pred = Rep("P", predicateBuilder...)
+			// Prove P = x0*B + x1*B + ... + xN*B
+			prover := pred.Prover(suite, sval, pval, nil)
+			proof, err = HashProve(suite, "TEST", prover)
+			if err != nil {
+				b.Log(err.Error())
+				b.Fail()
+			}
+		})
+
+		b.Run(suite.String()+"/ProofVerify", func(b *testing.B) {
+			verifier := pred.Verifier(suite, pval)
+			err = HashVerify(suite, "TEST", verifier, proof)
+			if err != nil {
+				b.Log(err.Error())
+				b.Fail()
+			}
+		})
```

Comment:
These are missing for loops in the subbenchmarks

Reply:
Fixed !